doc/memos/rdm4.md

```diff
@@ -0,0 +1,85 @@
+RIOT Maintainers Community
+Developer Memo: 4
+Author(s): Martine Lenders
+Date: January 2017
+
+Writing tests
+=============
+
+## Abstract
+This memo describes how to write tests for RIOT.
+
+
+## Status
+This document is a product of the community of RIOT maintainers, and aims to 
+represent the consensus within this community.
+The content of this document is licensed with a Creative Commons CC-BY-SA 
+license.
+
+## Terminology
+This memo uses the following terminology and acronyms:
+
+- embUnit: a unittest framework for C that is used by RIOT to write unittests.
+- pexpect: a python library to interact with the stdio of command-line
+  application.
+- CI system: CI = Continuous Intergration, platform for automated building and
+  testing of provided changes.
+
+## 1. Introduction
+At the moment of writing there are two strategies for testing: providing a
+unittest suite in [`tests/unittests/`][unittests] or a stand-alone test
+application in `tests/` directly. This document provides some recommendations
+on where to put a test and what the minimum requirements for those are.
+
```

Comment:
would it make sense to indicate at what point (level of granularity) one should consider providing tests?
Something like "if thou providest a new feature, thou shalt provide a test".
Then, if providing tests are to be provided, then the question is indeed unit test vs test in example.